README.md

```diff
@@ -1,223 +1,350 @@
+      this library as a proxy in conjunction with a Bitcoin Core node (with the Template Provider
+      logic) to do so.
+    * A miner who runs a mining farm with SV1-compatible mining firmware mining to a Sv2-compatible
+      pool, who wants to gain some of the security and efficiency improvements that Sv2 offers over
+      Stratum V1 (Sv1) (by using Extended Channels). This Sv1<->Sv2 miner proxy does not support
+      all the features of Sv2, therefore it should be used as a temporary measure before completely
+      upgrading to Sv2-compatible firmware.
+    * A pool that wants to build a Sv2 compatible pool server, complete with all Sv2 channel
+      support (Standard, Group, and Extended channels).
+    * A pool that wants to begin supporting Sv2 but is not yet ready to completely overhaul their
+      existing Sv1 server can use these libraries to construct a Sv2<->Sv1 proxy server. This proxy
+      server lives on the upstream pool end under the pool's control, between the downstream
+      devices on the miner's side (either a miner's proxy or the mining devices themselves), and
+      the most upstream Sv1 pool server. This most upstream Sv1 pool server is likely the already
+      existing pool server logic. In this way, pools can begin supporting Sv2 without waiting to
+      redo their entire server infrastructure to be Sv2 compatible only. This Sv2<->Sv1 pool proxy
+      will only allow the pool to support Standard channels (not Group or Extended), therefore this
+      should be used as a temporary measure before completely upgrading to a Sv2-only pool.
+    * A Sv2- or Sv1-compatible pool that wants to offer job negotiation capabilities to their
+      customers (miner's mining to their pool) can use this library to self-host a Job Negotiator
+      and Template Provider (a minimum amount of modification is still required).
+* Make this primitives available also to non Rust user.
+
+## 2. Structure
+```
+stratum
+  └─ examples
+  │   └─ interop-cpp
+  │   └─ interop-cpp-no-cargo
+  │   └─ ping-pong-with-noise
+  │   └─ ping-pong-without-noise
+  │   └─ sv1-client-and-server
+  │   └─ sv2-proxy
+  │   └─ template-provider-test
+  └─ experimental
+  │   └─ coinbase-negotiator
+  └─ protocols
+  │   └─ v1
+  │   └─ v2
+  │       └─ binary-sv2
+  │       │   └─ binary-sv2
+  │       │   └─ no-serde-sv2
+  │       │   └─ serde-sv2
+  │       └─ codec-sv2
+  │       └─ const-sv2
+  │       └─ framing-sv2
+  │       └─ noise-sv2
+  │       └─ roles-logic-sv2
+  │       └─ subprotocols
+  │       │   └─ common-messages
+  │       │   └─ job-negotiation
+  │       │   └─ mining
+  │       │   └─ template-distribution
+  │       └─ sv2-ffi
+  └─ roles/v2
+  │   └─ mining-proxy
+  │   └─ pool
+  │   └─ test-utils
+  │   │   └─ mining-device
+  │   │   └─ pool
+  └─ test
+  └─ utils
+      └─ buffer
+      └─ network-helpers
+```
+
+This workspace is divided in 6 macro-areas:
+1. `protocols`: Core Stratum V2 and V1 libraries.
```

Comment:
Maybe `Stratum V2 and V1 libraries.` is more clear than `Core Stratum V2 and V1 libraries.`

First things that I think when I read it is what does `Core` stand for?